python/message.py

```diff
@@ -22,7 +22,8 @@ def __init__(self, *args, **kwargs):
         UserDict.__init__(self,*args, **kwargs)
         self.context = {}
         # Assign it an ID
-        self.data['@id'] = str(uuid.uuid4())
+        if '@id' not in self.data:
+            self.data['@id'] = str(uuid.uuid4())
```

Comment:
Good idea, I think this was causing a JS error when looking at the routes and messages for their @id